.github/workflows/nightly-benchmark-overhead.yml

```
@@ -18,13 +18,19 @@ jobs:
           path: gh-pages
       - name: copy results from gh-pages branch
         run: |
-          rsync -avv gh-pages/benchmark-overhead/results/ benchmark-overhead/results/ && rm -rf gh-pages
+          rsync -avv gh-pages/benchmark-overhead/results/ benchmark-overhead/results/
       - name: run tests
+        working-directory: benchmark-overhead
         run: ./gradlew test
+      - name: inspect the results dir
         working-directory: benchmark-overhead
+        run: ls -lR results
+      - name: copy results back to gh-pages branch
+        run: rsync -avv benchmark-overhead/results/ gh-pages/benchmark-overhead/results/ && rm -rf benchmark-overhead/results
```

is the `rm -rf` part needed? (not that it's hurting..)
```suggestion
        run: rsync -avv benchmark-overhead/results/ gh-pages/benchmark-overhead/results/
```

---

Wasn't that because of this issue: https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/3973 ?

---

I have absolutely encountered problems with some of the GH actions when you have local changes. Some of them try and change branches and can't because of untracked changes. Removing the results like this helps to avoid those conditions. I'm hesitant to remove it...all the stars have to be aligned for it to work. ⭐ 😁